### ISSUE TITLE
VolumeSnapshot always in CreatingSnapshot phase

### ISSUE THREAD

**Issue by knkarthik** [kubernetes-sigs/aws-ebs-csi-driver#1065]
/kind bug

**What happened?**
I'm having the same issue as https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/871 even though my namespacing is correct for CR/CRB and SA ( I installed aws-ebs-csi-driver via Helm). 
I'm following the [volume snapshot examples](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/tree/master/examples/kubernetes/snapshot) and volume snapshots just shows:

```Events:
  Type    Reason            Age    From                 Message
  ----    ------            ----   ----                 -------
  Normal  CreatingSnapshot  3m47s  snapshot-controller  Waiting for a snapshot default/ebs-volume-snapshot to be created by the CSI driver.
```

**What you expected to happen?**
VolumeSnapshot should've been created.

**How to reproduce it (as minimally and precisely as possible)?**
1. Install aws-ebs-csi-driver via Helm
2. Install VolumeSnapshots CRDs:
  ```
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml
 ```
3.  Install Volume Snapshot controller
```
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml
```
4. Follow [volume snapshot examples](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/tree/master/examples/kubernetes/snapshot)

**Anything else we need to know?**:
I also tried following https://aws.amazon.com/blogs/containers/using-ebs-snapshots-for-persistent-storage-with-your-eks-cluster/ but hit the same issue. In addition, cleaning up of resources get stuck because of all the finalizers.

**Environment**
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:38:26Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.7-eks-d88609", GitCommit:"d886092805d5cc3a47ed5cf0c43de38ce442dfcb", GitTreeState:"clean", BuildDate:"2021-07-31T00:29:12Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}
```
- Driver version: v1.2.0

**Comment by nirmalaagash** [comment]
@knkarthik Did you enable external-snapshotter sidecar in your helm chart while installing?

**Comment by knkarthik** [comment]
@nirmalaagash Isn't it enabled by default according to https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/fff6f53afbbe2138ecd8711aa38fa6c67b7698de/charts/aws-ebs-csi-driver/values.yaml#L28? Do I have to do anything extra? I saw enabling snapshots and resize in https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html#helm as well but I don't see the corresponding keys in values files.

Anyways I just did a clean install and it looks to be working now.
